kubernetes/cray-tftp/templates/nf_nat_tftp.yaml

```diff
@@ -22,12 +22,15 @@ ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
 OTHER DEALINGS IN THE SOFTWARE.
 */}}
 {{- $baseChartValues := index .Values "cray-service" -}}
-# Purpose: The purpose of this daemonset is to allow the nf_nat_tftp module
-# to be loaded by the running kernel on all nodes that are candidates for
-# accepting ingress traffic into the network. Typically, this includes
-# worker nodes, but can include master nodes if the master taint is
-# removed. In BGP mode, this means any node that can serve as a speaker
-# should load this module.
+# Purpose: The nf_nat_tftp kernel module is required for cray-tftp pods to work on a node.
+# The purpose of this daemonset is to allow the nf_nat_tftp module  to be loaded by
+# the running kernel on all nodes running cray-tftp that are candidates for accepting
+# ingress traffic into the network. In BGP mode, this means any node that can serve as a
+# speaker should load this module.
+#
+# Because the cray-tftp pods only run on worker NCNs, this daemonset is only run on worker
+# NCNs as well. If the master taint was removed, allowing the cray-tftp pods to run on master NCNs,
```

Comment:
It would take more than the master taint being removed for this to work on master nodes.   We would also need to change the switch configs to have them peer with the master nodes.   Even with speakers running on the masters, nothing is peering with them so no LB traffic will ever go to the masters.

Reply:
Let me see if I can update the comment to capture this. I hesitate just to remove the comment and flatly state that it is only worker nodes, since I'd like to retain what was discussed as part of this ticket.

Reply:
So if the master taint was removed, yes, both the cray-tftp-modprobe would run on the masters and the cray-tftp pods could end up on the masters.   But the ingress traffic will still only go to the workers.

Reply:
One suggestion is to have this be a chart variable which is set to not run on master nodes by default.    Then in the future, if we ever decide we want to sending ingress to the master nodes (I really don't see us ever doing this), then it would just be a manifest change.   That's what MetalLB does.

Reply:
Thank you for the suggestion. I looked at how they do it and we could certainly do the same here. But I think to minimize the complexity of the changes in this PR, and given the unlikelihood of it being needed, I'll keep it as-is. But it's good to be aware of this as an option in the future, if we do need it.